src/Microsoft.AspNetCore.Mvc.DataAnnotations/Internal/MvcDataAnnotationsMvcOptionsSetup.cs

```diff
@@ -11,21 +11,28 @@ namespace Microsoft.AspNetCore.Mvc.DataAnnotations.Internal
     /// <summary>
     /// Sets up default options for <see cref="MvcOptions"/>.
     /// </summary>
-    public class MvcDataAnnotationsMvcOptionsSetup : ConfigureOptions<MvcOptions>
+    public class MvcDataAnnotationsMvcOptionsSetup : IConfigureOptions<MvcOptions>
     {
+        private IServiceProvider _serviceProvider;
+
         public MvcDataAnnotationsMvcOptionsSetup(IServiceProvider serviceProvider)
```

Comment:
We should use ctor overloads for this instead of using GetService:

```C#
public MvcDataAnnotationsMvcOptionsSetup(IValidationAttributeAdapterProvider validationAttributeAdapterProvider) : this (validationAttributeAdapterProvider, null)

public MvcDataAnnotationsMvcOptionsSetup(IValidationAttributeAdapterProvider validationAttributeAdapterProvider, IStringLocalizerFactory localizerFactory)
{
 ... 
}